### ISSUE TITLE
Contact picture not showing when receiving MMS

### ISSUE THREAD

**Issue by jrcast** [SMSSecure/SMSSecure#67]
You can see the screenshot. It should be pretty self explanatory.

Thanks!!!

![screenshot_2015-04-04](https://cloud.githubusercontent.com/assets/9587027/6995722/f9594376-db0e-11e4-90f1-d7163fb2e3b7.jpg)

**Comment by BastienLQ** [comment]
I think the bug is also in TextSecure.

**Comment by pR0Ps** [comment]
Upstream: https://github.com/WhisperSystems/TextSecure/issues/2532

**Comment by vdveer** [comment]
Sorry, I did not search for existing issues, as I entered this issue on a mobile device I just forgot. As for my comments in the other issue, I think this is not an upstream issue as unsecured MMS don't seem to be affected. Also when your sim has localnumber stored all seems to go well. I tried to find a hint in the code where things go wrong, see post from duplicate issue:

When TelephoneManager.getLine1Number has no value, and no verified phonenumber in SMSSecurePreference is available (I am guessing always since it is not required in this TextSecure fork), the PduComposer for the MMS message will not put a PduHeader.FROM value in the encrypted pdu as it is not set in the original. This leads to an anonymous incoming message (but since the outer pdu will supply a correct value I am guessing it will be saved with the correct threadid).

I have tried to check the MMSDownloadJob and fix the PduHeader value if empty to copy the header from the original pdu, but it doesn't seem to work. Probably I am not reading the code right, so if someone can hint a solution (if it is indeed I bug) I would be more than happy to help.
